src/lib/icon/icon-registry.ts

```diff
@@ -362,8 +356,20 @@ export class MatIconRegistry {
    * from it.
    */
   private _loadSvgIconSetFromConfig(config: SvgIconConfig): Observable<SVGElement> {
-    // TODO: Document that icons should only be loaded from trusted sources.
-    return this._fetchUrl(config.url).pipe(map(svgText => this._svgElementFromString(svgText)));
+    // If the SVG for this icon set has already been parsed, do nothing.
+    if (config.svgElement) {
+      return observableOf(config.svgElement);
+    }
+
+    return this._fetchUrl(config.url).pipe(map(svgText => {
+      // It is possible that the icon set was parsed and cached by an earlier request, so parsing
+      // only needs to occur if the cache is yet unset.
+      if (!config.svgElement) {
+        config.svgElement = this._svgElementFromString(svgText);
```

Comment:
This one could be shortened to `config.svgElement = config.svgElement || this._svgElementFromString(svgText)`.